controllers/storageclassclaim/storageclassclaim_controller.go

```diff
@@ -92,20 +124,296 @@ func (r *StorageClassClaimReconciler) Reconcile(ctx context.Context, request rec
+		response, err := providerClient.GetStorageClassClaimConfig(
+			r.ctx,
+			r.storageCluster.Status.ExternalStorage.ConsumerID,
+			r.storageClassClaim.Name,
+		)
+		if err != nil {
+			return reconcile.Result{}, fmt.Errorf("failed to get StorageClassClaim config: %v", err)
+		}
+		resources := response.ExternalResource
+		if resources == nil {
+			return reconcile.Result{}, fmt.Errorf("no configuration data recieved")
+		}
+
+		// Go over the received objects and operate on them accordingly.
+		for _, resource := range resources {
+			data := map[string]string{}
+			err = json.Unmarshal(resource.Data, &data)
+			if err != nil {
+				return reconcile.Result{}, fmt.Errorf("failed to unmarshal StorageClassClaim configuration response: %v", err)
+			}
+
+			switch resource.Kind {
+			case "Secret":
+				secret := &corev1.Secret{}
+				secret.Name = resource.Name
+				secret.Namespace = r.storageClassClaim.Namespace
+				_, err = controllerutil.CreateOrUpdate(r.ctx, r.Client, secret, func() error {
+					err := r.own(secret)
+					if err != nil {
+						return fmt.Errorf("failed to own Secret: %v", err)
+					}
+					m := map[string][]byte{}
+					for k, v := range data {
+						m[k] = []byte(v)
+					}
+					secret.Data = m
+					return nil
+				})
+				if err != nil {
+					return reconcile.Result{}, fmt.Errorf("failed to create or update secret %v: %s", secret, err)
+				}
+			case "StorageClass":
+				var storageClass *storagev1.StorageClass
+				if resource.Name == "cephfs" {
+					storageClass = r.getCephFSStorageClass(data)
+				} else if resource.Name == "ceph-rbd" {
+					storageClass = r.getCephRBDStorageClass(data)
+				}
+				err = r.createOrReplaceStorageClass(storageClass)
+				if err != nil {
+					return reconcile.Result{}, fmt.Errorf("failed to create or update StorageClass: %s", err)
+				}
+			}
+		}
+
+		// Readiness phase.
+		// Update the StorageClassClaim status.
+		r.storageClassClaim.Status.Phase = ocsv1alpha1.StorageClassClaimReady
+
+	}
+	// TODO(@rexagod): Deletion flow for SCC on Consumer side
+
 	return reconcile.Result{}, nil
 }
 
-func (r *StorageClassClaimReconciler) reconcileProviderPhases(ctx context.Context, storagecluster *v1.StorageCluster) (reconcile.Result, error) {
-	log := ctrllog.FromContext(ctx)
+func (r *StorageClassClaimReconciler) getCephFSStorageClass(data map[string]string) *storagev1.StorageClass {
+	pvReclaimPolicy := corev1.PersistentVolumeReclaimDelete
+	allowVolumeExpansion := true
+	storageClass := &storagev1.StorageClass{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      r.storageClassClaim.Name,
+			Namespace: r.storageClassClaim.Namespace,
+			Annotations: map[string]string{
+				"description": "Provides RWO and RWX Filesystem volumes",
+			},
+			Finalizers: []string{
+				StorageClassFinalizer,
+			},
+		},
+		ReclaimPolicy:        &pvReclaimPolicy,
+		AllowVolumeExpansion: &allowVolumeExpansion,
```

Comment:
```suggestion
ReclaimPolicy:        &corev1.corev1.PersistentVolumeReclaimDelete,
AllowVolumeExpansion: &allowVolumeExpansion,
```
You don't need a variable for `corev1.PersistentVolumeReclaimDelete` isn't it?

Reply:
Need a variable for this: https://stackoverflow.com/questions/35146286/find-address-of-constant-in-go